### PR TITLE
Use the Xunit assert library instead of CoreCLRTestLibrary for most asserts

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Asserts not provided by Xunit are provided by an AssertExtensions class.

Tests that reference System.Private.Corelib directly will use a polyfill implementation based off the old CoreCLRTestLibrary asserts.

All assert methods provided by CoreCLRTestLibrary have been changed to follow Xunit conventions (this is why the change is so noisy).

This contribues to our move towards more xunit-style tests in the src/tests tree, just with a custom runner (as part of the test merging initiative).

cc: @trylek for review

cc: @elinor-fung and @AaronRobinsonMSFT as the Interop team owns or has worked on most of the tests that are modified by this change.

### CODE REVIEW

File: src/tests/Common/Assert.cs [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        internal static void HandleFail(string assertionName, string message)
+        {
+            throw new XunitException(assertionName + ": " + message);
+        }
+
+
+        [Obsolete("Did you mean to call Assert.Equal()")]
+        public static new bool Equals(Object o1, Object o2)
+        {
+            Assert.True(false, "Don\u2019t call this.");
```
Comment (AaronRobinsonMSFT) [inline src/tests/Common/Assert.cs:207]:
```suggestion
            Assert.True(false, "Don't call this.");
```

File: src/tests/Common/Assert.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This file is used to provide some basic Assert functionality for assemblies that directly reference System.Private.CoreLib
+// and not the ref pack.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
```
Comment (AaronRobinsonMSFT) [inline src/tests/Common/Assert.cs:7]:
```suggestion
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by trylek — APPROVED** [review]
Looks great, thank you!

File: src/tests/Interop/ReadMe.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ There should be no more than **1** project type per folder (i.e. a folder can co
 
 Ancillary source assets for all tests should be located in `Interop/common` and can be easily added to all managed tests via the `Interop.settings.targets` file or native tests via `Interop.cmake`.
 
-A common pattern for testing is using the `Assert` utilities. This class is part of the `CoreCLRTestLibrary` which is included in all test projects by the `Interop.settings.targets` import. In order to use, add the following `using TestLibrary;` in the relevant test file.
+A common pattern for testing is using the `Assert` utilities. This class is part of the `CoreCLRTestLibrary` which is included in all test projects by the `Interop.settings.targets` import. In order to use, add the following `using Xunit;` in the relevant test file.
```
Comment (elinor-fung) [inline src/tests/Interop/ReadMe.md:13]:
```suggestion
A common pattern for testing is using xUnit's `Assert` utilities. These utilities can be referenced via `CoreCLRTestLibrary` which is included in all test projects by `Directory.Build.targets` in this directory. In order to use, add the following `using Xunit;` in the relevant test file.
```

**Review by elinor-fung — APPROVED** [review]
Would it be useful to add a bullet about test utilities / asserts to https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/coreclr/test-configuration.md#adding-test-guidelines?

File: src/tests/Interop/ReadMe.md (hunk 3 above)
Comment (trylek) [reply src/tests/Interop/ReadMe.md]: Fixed in latest commit.

**Comment by trylek** [comment]
The bugs are unrelated and known (https://github.com/dotnet/runtime/issues/61237, https://github.com/dotnet/runtime/issues/60152), merging in.

**Comment by jkoritzinsky** [comment]
@tylek I think there's a conflicting change that got merged in before this one. Can you check the Pri0 test build and make sure it isn't broken? (I'm not near my computer otherwise I'd do it myself)

**Comment by EgorBo** [comment]
Is failure like this https://dev.azure.com/dnceng/public/_build/results?buildId=1458028&view=logs&jobId=e2abd64e-6d96-553c-c262-deb1bd067c99&j=e2abd64e-6d96-553c-c262-deb1bd067c99&t=1849d1b9-84d9-551e-f982-c77c09278369 related?
`CoreCLR Common Pri0 Test Build AnyOS AnyCPU checked failed`
```
WeakReferenceTest.cs(260,20): error CS0117: 'Assert' does not contain a definition for 'IsNull' 
```

**Pull request merged**